src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/DeploymentConfigCompare.tsx

```diff
@@ -0,0 +1,369 @@
+            Promise.all([
+                getAppEnvDeploymentConfig({
+                    appName,
+                    envName: envName || '',
+                    configType,
+                    identifierId,
+                    pipelineId,
+                }),
+                getAppEnvDeploymentConfig({
+                    appName,
+                    envName: compareWith || '',
+                    configType: compareWithConfigType,
+                    identifierId: chartRefId || compareWithIdentifierId,
+                    pipelineId: compareWithPipelineId,
+                }),
+            ]),
+        [
+            appName,
+            compareWith,
+            compareWithConfigType,
+            configType,
+            identifierId,
+            compareWithIdentifierId,
+            pipelineId,
+            compareWithPipelineId,
+            chartRefId,
+        ],
+        !!configType && !!compareWithConfigType,
+    )
+
+    // Generate the deployment configuration list for the environments using comparison data
+    const appEnvDeploymentConfigList = useMemo(() => {
+        if (!comparisonDataLoader && comparisonData) {
+            const currentData = comparisonData[0].result
+            const compareData = comparisonData[1].result
+
+            const configData = getAppEnvDeploymentConfigList(currentData, compareData, path, params, search, sortOrder)
+            return configData
+        }
+
+        return null
+    }, [comparisonDataLoader, comparisonData, sortOrder])
+
+    // SELECT PICKER OPTIONS
+    /** Compare Environment Select Picker Options  */
+    const compareEnvironmentSelectorOptions: OptionsOrGroups<
+        SelectPickerOptionType,
+        GroupBase<SelectPickerOptionType>
+    > = [
+        {
+            label: BASE_CONFIGURATIONS.name,
+            value: '',
+        },
+        {
+            label: 'Environments',
+            options: environments.map(({ name }) => ({
+                label: name,
+                value: name,
+            })),
+        },
+        {
+            label: 'Default values',
+            options:
+                currentEnvOptions?.defaultVersions.map(({ chartRefId: _chartRefId, chartVersion }) => ({
+                    label: `v${chartVersion} (Default)`,
+                    value: _chartRefId,
+                })) || [],
+        },
+    ]
+
+    /**
+     * Get the configuration type options for the current or compare environment.
+     *
+     * @param isCompare - If true, compare env data will be used.
+     * @returns The configuration type options for the select picker.
+     */
+    const getEnvironmentConfigTypeOptions = (
+        isCompare?: boolean,
+    ): OptionsOrGroups<SelectPickerOptionType, GroupBase<SelectPickerOptionType>> => [
+        {
+            label: 'Published only',
+            value: AppEnvDeploymentConfigType.PUBLISHED_ONLY,
+            description: 'Configurations that will be deployed in the next deployment',
+        },
```

Comment:
suggestion: the options part can be moved to util function

Reply:
not clear

Reply:
The options can be abstracted to a separate util file thereby reducing the overall file size